lib/internal/fs/glob.js

```diff
@@ -224,10 +210,10 @@ class Glob {
 
   globSync() {
     ArrayPrototypePush(this.#queue, { __proto__: null, path: '.', patterns: this.#patterns });
-    while (this.#queue.length > 0) {
-      const item = ArrayPrototypePop(this.#queue);
-      for (let i = 0; i < item.patterns.length; i++) {
-        this.#addSubpatterns(item.path, item.patterns[i]);
+    while (this.#queue.length) {
+      const { path, patterns } = ArrayPrototypePop(this.#queue);
+      for (const pattern of patterns) {
```

Comment:
it is unsafe to iterate on patterns, that is why an index is used

Reply:
Agreed